doc/spec_InterruptControllerHandler.dox

```diff
@@ -0,0 +1,91 @@
+
+## Introduction ##
+
+This document describes the concept of interrupt handlers in backends, their implementation and their interface to firmware, i.e. their description in the map file.
+
+### Context ###
+
+The current implementation INTERRUPTc:i (with c being the interrupt controller number and i being the interrupt) is wrong on several levels.
+* The highest level is always associated with a hardware interrupt. If there is one interrupt controller, this is correct. If there is no interrupt controller, it is already wrong. It currently is INTERRUPT0:i, but should be INTERRUPTi. (e.g. with interrupt controller INTERRUPT0:5 is on HW interrupt 0, while without it is on HW interrupt 5, which is inconsistent).
+* Nested interrupts cannot be described at all.
+
+This specification describes the hierarchical interrupt scheme that solves these issues.
+
+## A. Definitions ##
+
+- hwInterruptID is an integer which identifies the primary hardware interrupt
+- qualifiedID is the ID including all the parent hierarcy. It can be evaluated on any level.
+  Example: INTERRUPT2:3:4 on a nested interrupt controller
+  - 2 is the qualifiedID of the parent interrupt controller (identical to the HW interrupt ID, see \ref interruptHandler_B1_1).
+  - 2:3 is the qualified ID of the nested interrupt controller (identical to the interrupt generated by interrupt controller 2)
+  - 2:3:4 is the qualified ID of the interrupt generated by controller 2:3
+- INTERRUPTi[:j[:k[...]]] means any level of nesting
+
+- The interrupt controller resides in firmware
+- The InterruptControllerHandler is the software class that interacts with the according interrupt controller (resolve interrupt lane and perform handshake)
+- Interrupt handler and interrupt handling are generic. It can be the handling of a HW interrupt lane or inside a controller.
+- HW interrupt is used for the primary interrupt where no interrupt controller is involved that is handled by software (for instance Message Signaled PCIe Interrupts are considered as HW interrupt)
+
+## B. Behavioural specification
+
+- B1. Interrupts and interrupt controllers are identified with a numerical ID (unsigned integer)
+  - \anchor interruptHandler_B1_1 B1.1 The controller ID is always identical to the parent interrput (e.g. controller 1 is behind hardware interrupt 1)
+- B2. To identify interrupts resolved by a controller, their IDs are appended to the parent interrupt in a hierarchical way. This allows to nest interrupt controllers.
+- B3. In string form (e.g. in the map file and this document), a colon ":" is used as hierarchy separator.
+  - B3.1 Programmatic implementations can vary from this (e.g. in C++ the interrupt can be descibed as a list/vector of integers)
+- B4. It is always possible to get an accessor for all interrupt hierarchy levels.
+  - B4.1 An accessor with the name INTERRUPTk[:l[:m[...]]] can be acquired, even if there is no map file entry for it.
+- B5. Backend implementations (NumericAddressedBackend)
```

Comment:
Is this still behaviour, or a new section for the implementation? This depends on the aim of this document, so I am really unsure...